test/libsolidity/semanticTests/various/eof/create_calldata.sol

```diff
@@ -0,0 +1,27 @@
+contract C {
+	bytes public s;
+	constructor(uint256 x, bytes memory b) {
+		// On EOF msg.data contains constructor arguments, while on legacy it's empty
+		// (arguments are appended to the code).
+		s = msg.data;
+		assert(msg.data.length == 288);
+		uint y;
+		bytes memory d;
+		(y, d) = abi.decode(msg.data, (uint256, bytes));
+		assert(x == y);
+		assert(b.length == d.length);
+		for (uint i = 0; i < b.length; ++i)
+			assert(b[i] == d[i]);
+	}
+}
+// ====
+// bytecodeFormat: >=EOFv1
+// ----
+// constructor(): 42, 0x20, 0xc0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6 ->
+// gas irOptimized: 68239
+// gas irOptimized code: 69000
+// gas legacy: 78076
+// gas legacy code: 90200
+// gas legacyOptimized: 68321
+// gas legacyOptimized code: 64600
+// s() -> 0x20, 0x0120, 0x2a, 0x20, 0xc0, 1, 2, 3, 4, 5, 6
```

Comment:
Please remove the costs.


While at it, I'd also use the same format for matching numbers to make it easier to see at a glance that they do match. The default formatting by isoltest is not always the greatest, but it will still accept a different one as long as the value is the same:
```suggestion
// constructor(): 42, 0x20, 0xc0, 1, 2, 3, 4, 5, 6 ->
// s() -> 0x20, 0x0120, 42, 0x20, 0xc0, 1, 2, 3, 4, 5, 6
```

Reply:
done